### ISSUE TITLE
Generating a property type as a Enumeration from a foreign key 

### ISSUE THREAD

**Issue by James00-Fast** [sjh37/efreversepoco#748]
Has anyone experienced this issue ? 

The relationship from 'Common_TestABC.Common_Status' to 'Common_Status.Common_TestAbcs' with foreign key properties {'StatusId' : StatusTest} cannot target the primary key {'StatusId' : int} because it is not compatible. Configure a principal key or a set of foreign key properties with compatible types for this relationship.

We are trying to generate a class from SQL. The table TestABC has a foreign key reference called StatusId to the Status table. Every time we startup the solution we receive the above error. This happens for any table we attempt. The idea is to be able to use the generation in a way TestABC.StatusId = Status (Enum). Rather than creating our own Enums by hand. 

I have, among others, tried EnumerationSettings & enumDefinitions. Running out of options. The .tt looks fine. 
**Is this a known issue ?** 

```cs
new EnumerationSettings
{
    Name       = "StatusTest",
    Table      = "Common.Status",
    NameField  = "Description",
    ValueField = "StatusId"
}

enumDefinitions.Add(new EnumDefinition { Schema = "Common", Table = "TestABC", Column = "StatusId", EnumType = "StatusTest" });
```

**Comment by James00-Fast** [comment]
Simple table layout example. 

![image](https://user-images.githubusercontent.com/53894549/160407605-d6f05100-a1c2-4a79-bd1c-05a22658ab98.png)

![image](https://user-images.githubusercontent.com/53894549/160407882-061347ff-4914-4d4f-9669-34fc350adc5d.png)

**Comment by sjh37** [comment]
Sorry for the error! 
In class Common_Status, change the PK to also be the enum as below:

```cs
// Status
public class Common_Status
{
    public StatusTest StatusId { get; set; } // StatusId (Primary key)   <<<========== now an enum
    public string Description { get; set; } // Description (length: 50)

    // Reverse navigation

    /// <summary>
    /// Child Common_TestAbcs where [TestABC].[StatusId] point to this entity (FkNameAbc)
    /// </summary>
    public virtual ICollection<Common_TestAbc> Common_TestAbcs { get; set; } // TestABC.FkNameAbc

    public Common_Status()
    {
        Common_TestAbcs = new List<Common_TestAbc>();
    }
}
```

I will mention this case when the fix is in.

**Comment by sjh37** [comment]
SQL to generate tables above:

```sql
CREATE TABLE Common.[Status]
(
    StatusId INT IDENTITY(1, 1) NOT NULL,
    [Description] NVARCHAR(50) NOT NULL,
    CONSTRAINT PK_Status
        PRIMARY KEY CLUSTERED (StatusId ASC)
);
GO
CREATE TABLE Common.TestABC
(
    TestABCId INT IDENTITY(1, 1) NOT NULL,
    StatusId INT NOT NULL,
    FieldA VARCHAR(1) NULL,
    FieldB BIT NULL,
    CONSTRAINT PK_TestABC
        PRIMARY KEY CLUSTERED (TestABCId ASC),
    CONSTRAINT FkNameAbc
        FOREIGN KEY (StatusId)
        REFERENCES common.[Status] (StatusId)
);
GO
INSERT INTO Common.[Status] ([Description])
VALUES ('Ok'),('Info'),('Warning'),('Error'),('Critical');
GO
```

**Comment by James00-Fast** [comment]
@sjh37 Thank you very much ! Your fast response and effort is appreciated. This worked for us. Added an extra enumDefinitions liked you mentioned 🙌.

**Comment by sjh37** [comment]
Now released in 3.6.1

**Comment by NeilN1** [comment]
I have a similar issue running v3.6.1. Is there still a manual fix to be done?

Tables:

```sql
CREATE TABLE [dbo].[MemberSkill](
	[MemberSkillId] [uniqueidentifier] NOT NULL,
	[CreatedOn] [datetime] NOT NULL,
	[CreatedBy] [nvarchar](100) NOT NULL,
	[ModifiedOn] [datetime] NULL,
	[ModifiedBy] [nvarchar](100) NULL,
	[ContactId] [uniqueidentifier] NOT NULL,
	[SkillId] [uniqueidentifier] NOT NULL,
	[EffectiveDate] [datetime] NULL,
	[ExpiryDate] [datetime] NULL,
	[SkillStatusId] [int] NULL,
	[SkillStatusDate] [date] NULL,
 CONSTRAINT [PK_MemberSkill] PRIMARY KEY CLUSTERED 
(
	[MemberSkillId] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
GO

ALTER TABLE [dbo].[MemberSkill]  WITH CHECK ADD  CONSTRAINT [FK_MemberSkill_Skill] FOREIGN KEY([SkillId])
REFERENCES [dbo].[Skill] ([SkillId])
GO

ALTER TABLE [dbo].[MemberSkill] CHECK CONSTRAINT [FK_MemberSkill_Skill]
GO

ALTER TABLE [dbo].[MemberSkill]  WITH CHECK ADD  CONSTRAINT [FK_MemberSkill_SkillStatus] FOREIGN KEY([SkillStatusId])
REFERENCES [dbo].[SkillStatus] ([SkillStatusId])
GO

ALTER TABLE [dbo].[MemberSkill] CHECK CONSTRAINT [FK_MemberSkill_SkillStatus]
GO

CREATE TABLE [dbo].[SkillStatus](
	[SkillStatusId] [int] NOT NULL,
	[Description] [nvarchar](100) NOT NULL,
	[DisplayOrder] [int] NOT NULL,
 CONSTRAINT [PK_SkillStatus] PRIMARY KEY CLUSTERED 
(
	[SkillStatusId] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
GO
```

Autogenerated POCOs and Enum
```c#
// MemberSkill
    public partial class MemberSkill
    {
        public Guid MemberSkillId { get; set; } // MemberSkillId (Primary key)
        public DateTime CreatedOn { get; set; } // CreatedOn
        public string CreatedBy { get; set; } // CreatedBy (length: 100)
        public DateTime? ModifiedOn { get; set; } // ModifiedOn
        public string ModifiedBy { get; set; } // ModifiedBy (length: 100)
        public Guid ContactId { get; set; } // ContactId
        public Guid SkillId { get; set; } // SkillId
        public DateTime? EffectiveDate { get; set; } // EffectiveDate
        public DateTime? ExpiryDate { get; set; } // ExpiryDate
        public SkillStatusEnum? SkillStatusId { get; set; } // SkillStatusId
        public DateTime? SkillStatusDate { get; set; } // SkillStatusDate

        // Foreign keys

        /// <summary>
        /// Parent Skill pointed by [MemberSkill].([SkillId]) (FK_MemberSkill_Skill)
        /// </summary>
        public virtual Skill Skill { get; set; } // FK_MemberSkill_Skill

        /// <summary>
        /// Parent SkillStatus pointed by [MemberSkill].([SkillStatusId]) (FK_MemberSkill_SkillStatus)
        /// </summary>
        public virtual SkillStatus SkillStatus { get; set; } // FK_MemberSkill_SkillStatus

        public MemberSkill()
        {
            InitializePartial();
        }

        partial void InitializePartial();
    }

    // SkillStatus
    public partial class SkillStatus
    {
        public int SkillStatusId { get; set; } // SkillStatusId (Primary key)
        public string Description { get; set; } // Description (length: 100)
        public int DisplayOrder { get; set; } // DisplayOrder

        // Reverse navigation

        /// <summary>
        /// Child MemberSkills where [MemberSkill].[SkillStatusId] point to this entity (FK_MemberSkill_SkillStatus)
        /// </summary>
        public virtual ICollection<MemberSkill> MemberSkills { get; set; } // MemberSkill.FK_MemberSkill_SkillStatus

        public SkillStatus()
        {
            MemberSkills = new List<MemberSkill>();
            InitializePartial();
        }

        partial void InitializePartial();
    }

    public enum SkillStatusEnum
    {
        Pending = 1,
        Proven = 2,
        Invalid = 3,
    }

```

The relationship from 'MemberSkill.SkillStatus' to 'SkillStatus.MemberSkills' with foreign key properties {'SkillStatusId' : SkillStatusEnum?} cannot target the primary key {'SkillStatusId' : int} because it is not compatible. Configure a principal key or a set of foreign key properties with compatible types for this relationship.
